src/shared/install.c

```diff
@@ -1766,9 +1772,14 @@ static int install_context_mark_for_removal(
                         return r;
 
                 r = install_info_traverse(scope, c, paths, i, SEARCH_LOAD|SEARCH_FOLLOW_CONFIG_SYMLINKS, NULL);
-                if (r == -ENOLINK)
+                if (r == -ENOLINK) {
+                        log_debug("Name %s leads to a dangling symlink, ignoring.", i->name);
+                        continue;
+                } else if (r == -ENOENT && i->auxiliary) {
+                        /* some unit specified in Also= or similar is missing */
+                        log_debug("Unit %s not found, ignoring.", i->name);
```

Comment:
nitpick 1: `log_debug_errno(r, …` would be cool...
nitpick 2: maybe say "`Auxiliary unit %s,…`", maybe as an explanation why we ignore it...